catalog/async/catalog-async-postingestplugin/src/test/groovy/org/codice/ddf/catalog/async/processingplugin/ProcessingPostIngestPluginTest.groovy

```diff
@@ -0,0 +1,410 @@
+        def responseProperties = [a: _]
+
+        def input = Mock(DeleteResponse) {
+            getRequest() >> Mock(DeleteRequest) {
+                getProperties() >> properties
+            }
+
+            getDeletedMetacards() >> [metacard]
+
+            getProperties() >> responseProperties
+        }
+
+        when:
+        def output = processingPostIngestPlugin.process(input)
+
+        then:
+        output == input
+
+        1 * processingFramework.submitDelete(_ as ProcessRequest<ProcessDeleteItem>) >> { ProcessRequest<ProcessDeleteItem> processDeleteRequest ->
+            assert processDeleteRequest.getProcessItems().size() == 1
+            assert processDeleteRequest.getProcessItems().get(0).getMetacard() == metacard
+            assert postProcessCompleteEntryAdded(processDeleteRequest.getProperties(), responseProperties)
+        }
+
+        where:
+        properties << [createFalseProcessingCompleteProperties(), createNotBooleanProcessingCompleteProperties(), [:]]
+    }
+
+    def 'test process DeleteResponse with no deleted metacards'() {
+        given:
+        def responseProperties = [a: _]
+
+        def input = Mock(DeleteResponse) {
+            getRequest() >> Mock(DeleteRequest) {
+                getProperties() >> properties
+            }
+
+            getDeletedMetacards() >> []
+
+            getProperties() >> responseProperties
+        }
+
+        when:
+        def output = processingPostIngestPlugin.process(input)
+
+        then:
+        output == input
+
+        0 * catalogFramework._
+        1 * processingFramework.submitDelete(_ as ProcessRequest<ProcessDeleteItem>) >> { ProcessRequest<ProcessDeleteItem> processDeleteRequest ->
+            assert processDeleteRequest.getProcessItems().isEmpty()
+            assert postProcessCompleteEntryAdded(processDeleteRequest.getProperties(), responseProperties)
+        }
+
+        where:
+        properties << [createFalseProcessingCompleteProperties(), createNotBooleanProcessingCompleteProperties(), [:]]
+    }
+
+    def 'test no process DeleteResponse'() {
+        when:
+        def output = processingPostIngestPlugin.process(input as DeleteResponse)
+
+        then:
+        output == input
+
+        0 * processingFramework._
+
+        where:
+        input << [null, Mock(DeleteResponse) {
+            getDeletedMetacards() >> null
+        }, Mock(DeleteResponse) {
+            getDeletedMetacards() >> []
+            getRequest() >> Mock(DeleteRequest) {
+                getProperties() >> createProcessingCompleteProperties()
+            }
+        }]
+    }
+
+    /*
+    other tests
+    */
+
+    def 'test catalogFramework.getResource Exceptions'() {
+        given:
+        final String sourceId
+        def metacard = Mock(Metacard) {
+            getSourceId() >> sourceId
+        }
+
+        def responseProperties = [a: _]
+
+        def input = Mock(CreateResponse) {
+            getRequest() >> Mock(CreateRequest) {
```

Comment:
❗️ Shouldn't we also test that the exceptions are handled properly by the `process(UpdateRequest)` method?